pkg/controller/util/node/controller_utils.go

```diff
@@ -99,7 +102,14 @@ func DeletePods(ctx context.Context, kubeClient clientset.Interface, pods []*v1.
 // pod status, updates it in the apiserver, and returns an error if it
 // encounters one.
 func SetPodTerminationReason(ctx context.Context, kubeClient clientset.Interface, pod *v1.Pod, nodeName string) (*v1.Pod, error) {
-	if pod.Status.Reason == nodepkg.NodeUnreachablePodReason {
+	var updatedCondition bool = false
+	if feature.DefaultFeatureGate.Enabled(features.JobBackoffPolicy) {
+		updatedCondition = utilpod.UpdatePodCondition(&pod.Status, &v1.PodCondition{
+			Type:   v1.EvictedFromUnhealtyNode,
+			Status: v1.ConditionTrue,
+		})
+	}
+	if !updatedCondition && pod.Status.Reason == nodepkg.NodeUnreachablePodReason {
```

Comment:
we should keep the reason (someone might be relying on it, even though we don't provide guarantees about it)

Reply:
ah true. Sounds good.